fhir-model/src/main/java/org/linuxforhealth/fhir/model/string/util/strategy/MaxBytesStringSizeControlStrategy.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.linuxforhealth.fhir.model.string.util.strategy;
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CodingErrorAction;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import org.linuxforhealth.fhir.model.string.util.StringSizeControlStrategyFactory.Strategy;
+
+/**
+ * Truncate the input String value to fit the input maxBytes(maximum bytes) size. 
+ */
+public class MaxBytesStringSizeControlStrategy implements StringSizeControlStrategy {
+
+    private static final Logger LOG = Logger.getLogger(MaxBytesStringSizeControlStrategy.class.getName());
+    
+    @Override
+    public String truncateString(String value, int maximumBytes) {
+        
+        LOG.fine("truncate input string " + value + ", to " + maximumBytes);
```

Comment:
if this gets called a lot, you might want to protect it with either 
A. an `if (isLoggable(Level.FINE))`; or
B. this lambda flavor `LOG.fine(() -> "truncate input string " + value + ", to " + maximumBytes);`